src/btree/bt_sync.c

```diff
@@ -110,6 +111,8 @@ __sync_file(WT_SESSION_IMPL *session, int syncop)
 		/* Write all dirty in-cache pages. */
 		flags |= WT_READ_NO_EVICT;
 		for (walk = NULL;;) {
+			if (txn->isolation == WT_ISO_READ_COMMITTED)
```

Comment:
A comment explaining the situation where this comes into play would be useful.

As discussed, it might make more sense to only get a snapshot once, rather than each step of the walk.